//
// PROJECT:         Aspia
// FILE:            category/category_cpu.proto
// LICENSE:         GNU Lesser General Public License 2.1
// PROGRAMMERS:     Dmitry Chapyshev (dmitry@aspia.ru)
//

syntax = "proto3";

option optimize_for = LITE_RUNTIME;

package aspia.proto;

message CPU
{
    message Features
    {
        bool has_fpu             = 1;
        bool has_vme             = 2;
        bool has_de              = 3;
        bool has_pse             = 4;
        bool has_tsc             = 5;
        bool has_msr             = 6;
        bool has_pae             = 7;
        bool has_mce             = 8;
        bool has_cx8             = 9;
        bool has_apic            = 10;
        bool has_sep             = 11;
        bool has_mtrr            = 12;
        bool has_pge             = 13;
        bool has_mca             = 14;
        bool has_cmov            = 15;
        bool has_pat             = 16;
        bool has_pse36           = 17;
        bool has_psn             = 18;
        bool has_clfsh           = 19;
        bool has_ds              = 20;
        bool has_acpu            = 21;
        bool has_mmx             = 22;
        bool has_fxsr            = 23;
        bool has_sse             = 24;
        bool has_sse2            = 25;
        bool has_ss              = 26;
        bool has_htt             = 27;
        bool has_tm              = 28;
        bool has_ia64            = 29;
        bool has_pbe             = 30;
        bool has_sse3            = 31;
        bool has_pclmuldq        = 32;
        bool has_dtes64          = 33;
        bool has_monitor         = 34;
        bool has_ds_cpl          = 35;
        bool has_vmx             = 36;
        bool has_smx             = 37;
        bool has_est             = 38;
        bool has_tm2             = 39;
        bool has_ssse3           = 40;
        bool has_cnxt_id         = 41;
        bool has_sdbg            = 42;
        bool has_fma             = 43;
        bool has_cx16            = 44;
        bool has_xtpr            = 45;
        bool has_pdcm            = 46;
        bool has_pcid            = 47;
        bool has_dca             = 48;
        bool has_sse41           = 49;
        bool has_sse42           = 50;
        bool has_x2apic          = 51;
        bool has_movbe           = 52;
        bool has_popcnt          = 53;
        bool has_tsc_deadline    = 54;
        bool has_aes             = 55;
        bool has_xsave           = 56;
        bool has_osxsave         = 57;
        bool has_avx             = 58;
        bool has_f16c            = 59;
        bool has_rdrand          = 60;
        bool has_hypervisor      = 61;
        bool has_syscall         = 62;
        bool has_xd_bit          = 63;
        bool has_mmxext          = 64;
        bool has_1gb_pages       = 65;
        bool has_rdtscp          = 66;
        bool has_intel64         = 67;
        bool has_3dnowext        = 68;
        bool has_3dnow           = 69;
        bool has_lahf            = 70;
        bool has_svm             = 71;
        bool has_lzcnt           = 72;
        bool has_sse4a           = 73;
        bool has_misalignsse     = 74;
        bool has_3dnow_prefetch  = 75;
        bool has_xop             = 76;
        bool has_wdt             = 77;
        bool has_fma4            = 78;
        bool has_fsgsbase        = 79;
        bool has_sgx             = 80;
        bool has_bmi1            = 81;
        bool has_hle             = 82;
        bool has_avx2            = 83;
        bool has_smep            = 84;
        bool has_bmi2            = 85;
        bool has_erms            = 86;
        bool has_invpcid         = 87;
        bool has_rtm             = 88;
        bool has_pqm             = 89;
        bool has_mpx             = 90;
        bool has_pqe             = 91;
        bool has_avx512f         = 92;
        bool has_avx512dq        = 93;
        bool has_rdseed          = 94;
        bool has_adx             = 95;
        bool has_smap            = 96;
        bool has_avx512ifma      = 97;
        bool has_pcommit         = 98;
        bool has_clflushopt      = 99;
        bool has_clwb            = 100;
        bool has_intel_pt        = 101;
        bool has_avx512pf        = 102;
        bool has_avx512er        = 103;
        bool has_avx512cd        = 104;
        bool has_sha             = 105;
        bool has_avx512bw        = 106;
        bool has_avx512vl        = 107;
        bool has_prefetchwt1     = 108;
        bool has_avx512vbmi      = 109;
        bool has_umip            = 110;
        bool has_pku             = 111;
        bool has_ospke           = 112;
        bool has_avx512vbmi2     = 113;
        bool has_gfni            = 114;
        bool has_vaes            = 115;
        bool has_vpclmulqdq      = 116;
        bool has_avx512vnni      = 117;
        bool has_avx512bitalg    = 118;
        bool has_avx512vpopcntdq = 119;
        bool has_rdpid           = 120;
        bool has_sgx_lc          = 121;
        bool has_avx512_4vnniw   = 122;
        bool has_avx512_4fmaps   = 123;
        bool has_ais             = 124;
        bool has_ais_e           = 125;
        bool has_rng             = 126;
        bool has_rng_e           = 127;
        bool has_lh              = 128;
        bool has_femms           = 129;
        bool has_ace             = 130;
        bool has_ace_e           = 131;
        bool has_ace2            = 132;
        bool has_ace2_e          = 133;
        bool has_phe             = 134;
        bool has_phe_e           = 135;
        bool has_pmm             = 136;
        bool has_pmm_e           = 137;
        bool has_parallax        = 138;
        bool has_parallax_e      = 139;
        bool has_overstress      = 140;
        bool has_overstress_e    = 141;
        bool has_tm3             = 142;
        bool has_tm3_e           = 143;
        bool has_rng2            = 144;
        bool has_rng2_e          = 145;
        bool has_phe2            = 146;
        bool has_phe2_e          = 147;
        bool has_ibs             = 148;
        bool has_skinit          = 149;
        bool has_lwp             = 150;
        bool has_tbm             = 151;
        bool has_perfctr_core    = 152;
        bool has_perfctr_nb      = 153;
        bool has_bpext           = 154;
        bool has_ptsc            = 155;
        bool has_mwaitx          = 156;
        bool has_extapic         = 157;
        bool has_longrun         = 158;
        bool has_lrti            = 159;
    }

    string brand_string      = 1;
    string vendor            = 2;
    uint32 stepping          = 3;
    uint32 model             = 4;
    uint32 family            = 5;
    uint32 extended_model    = 6;
    uint32 extended_family   = 7;
    uint32 brand_id          = 8;
    uint32 packages          = 9;
    uint32 physical_cores    = 10;
    uint32 logical_cores     = 11;
    Features features        = 12;
}
